src/core/server/plugins/discovery/plugin_manifest_parser.ts

```diff
@@ -157,26 +158,49 @@ export async function parseManifest(
     );
   }
 
-  if (
-    manifest.requiredOpenSearchPlugins !== undefined &&
-    (!Array.isArray(manifest.requiredOpenSearchPlugins) ||
-      !manifest.requiredOpenSearchPlugins.every((plugin) => typeof plugin === 'string'))
-  ) {
-    throw PluginDiscoveryError.invalidManifest(
-      manifestPath,
-      new Error(
-        `The "requiredOpenSearchPlugins" in plugin manifest for "${manifest.id}" should be an array of strings.`
-      )
-    );
+  if (manifest.requiredEnginePlugins !== undefined) {
+    for (const [pluginId, pluginVersion] of Object.entries(manifest.requiredEnginePlugins)) {
+      if (
+        typeof manifest.requiredEnginePlugins !== 'object' ||
+        typeof pluginId !== 'string' ||
+        typeof pluginVersion !== 'string'
+      ) {
+        throw PluginDiscoveryError.invalidManifest(
+          manifestPath,
+          new Error(
+            `The "requiredEnginePlugins" in plugin manifest for "${manifest.id}" should be a map of plugin name to its compatible version ranges.`
+          )
+        );
+      }
+    }
   }
 
   if (
-    Array.isArray(manifest.requiredOpenSearchPlugins) &&
-    manifest.requiredOpenSearchPlugins.length > 0
+    manifest.requiredEnginePlugins !== undefined &&
+    Object.keys(manifest.requiredEnginePlugins).length > 0
   ) {
     log.info(
-      `Plugin ${manifest.id} has a dependency on following OpenSearch plugin(s): "${manifest.requiredOpenSearchPlugins}".`
+      `Plugin ${manifest.id} has a dependency on following OpenSearch plugin(s): [${Object.keys(
+        manifest.requiredEnginePlugins
+      )}]`
     );
+
+    const invalidPluginVersions: string[] = [];
+    for (const [pluginName, versionRange] of Object.entries(manifest.requiredEnginePlugins)) {
+      if (!isOpenSearchPluginVersionRangeValid(versionRange)) {
+        invalidPluginVersions.push(pluginName);
+      }
+    }
+
+    if (invalidPluginVersions.length > 0) {
+      const pluginNames = invalidPluginVersions.join(', ');
+      throw PluginDiscoveryError.invalidManifest(
+        manifestPath,
+        new Error(
+          `The "requiredEnginePlugins" in plugin manifest for "${manifest.id}" has an invalid version range defined for the following plugins: ${pluginNames}`
+        )
+      );
+    }
   }
```

Comment:
This is untested and unlinted but am sure you get the jist:
```suggestion
  if ('requiredEnginePlugins' in manifest) {
    if (
      typeof manifest.requiredEnginePlugins !== 'object' ||
      !Object.entries(manifest.requiredEnginePlugins).every(
        ([pluginId, pluginVersion]) =>
          typeof pluginId === 'string' && typeof pluginVersion === 'string'
      )
    ) {
      throw PluginDiscoveryError.invalidManifest(
        manifestPath,
        new Error(
          `The "requiredEnginePlugins" in the plugin manifest for "${manifest.id}" should be a simple object that maps of plugin name to a version range.`
        )
      );
    }

    /* At this point it is guaranteed that `manifest.requiredEnginePlugins` is a simple object
     * of `string => string`, but possibly empty which is fine.
     */
    const invalidPluginVersions: string[] = [];
    for (const [pluginName, versionRange] of Object.entries(manifest.requiredEnginePlugins)) {
      log.info(`Plugin ${manifest.id} has a dependency on engine plugin: ${pluginName}@${versionRange}`);

      if (!isOpenSearchPluginVersionRangeValid(versionRange)) {
        invalidPluginVersions.push(`${versionRange} for ${pluginName}`);
      }
    }

    if (invalidPluginVersions.length > 0) {
      throw PluginDiscoveryError.invalidManifest(
        manifestPath,
        new Error(
          `The "requiredEnginePlugins" in the plugin manifest for "${manifest.id}" contains invalid version ranges: ${invalidPluginVersions.join(', ')}`
        )
      );
    }
  }
```

Reply:
Thanks for taking time to improve this.  I've updated the changes.